internal/controller/object/object.go

```diff
@@ -483,6 +488,105 @@ func getReferenceInfo(ref v1alpha2.Reference) (string, string, string, string) {
 	return apiVersion, kind, namespace, name
 }
 
+func (c *external) checkDeriveFromObject(observed *unstructured.Unstructured) (ready bool) {
+	conditioned := xpv1.ConditionedStatus{}
+	err := fieldpath.Pave(observed.Object).GetValueInto("status", &conditioned)
+	if err != nil {
+		c.logger.Debug("Got error while getting conditions from observed object, setting it as Unavailable", "error", err, "observed", observed)
+		return
+	}
+	if status := conditioned.GetCondition(xpv1.TypeReady).Status; status != v1.ConditionTrue {
+		c.logger.Debug("Observed object is not ready, setting it as Unavailable", "status", status, "observed", observed)
+		return
+	}
+	ready = true
+	return
+}
```

Comment:
nit: we try to avoid naked returns, as they make code more obscure.
```suggestion
func (c *external) checkDeriveFromObject(observed *unstructured.Unstructured) bool {
	conditioned := xpv1.ConditionedStatus{}
	if err := fieldpath.Pave(observed.Object).GetValueInto("status", &conditioned); err != nil {
		c.logger.Debug("Got error while getting conditions from observed object, setting it as Unavailable", "error", err, "observed", observed)
		return false
	}
	if status := conditioned.GetCondition(xpv1.TypeReady).Status; status != v1.ConditionTrue {
		c.logger.Debug("Observed object is not ready, setting it as Unavailable", "status", status, "observed", observed)
		return false
	}
	return true
}
```